File: src/dataset/python/triangle/gemini/GeminiFlash_triangle_v4.py
def equilateral(sides):
    if len(sides) != 3 or any(s <= 0 for s in sides):
        return False
    a, b, c = sides
    return a == b == c

def isosceles(sides):
    if len(sides) != 3 or any(s <= 0 for s in sides):
        return False
    a, b, c = sides
    return (a == b) or (a == c) or (b == c)

def scalene(sides):
    if len(sides) != 3 or any(s <= 0 for s in sides):
        return False
    a, b, c = sides
    return (a != b) and (a != c) and (b != c)

def triangle(sides):
    if len(sides) != 3:
        return False
    a, b, c = sides
    if any(s <= 0 for s in sides):
        return False
    return a + b > c and a + c > b and b + c > a

File: src/dataset/cpp/triangle_Exercism_cmccandless/gemini/GeminiFlash_triangle_v4.cpp
#include "triangle.h"
#include <algorithm>

namespace triangle
{
	void validate(double a, double b, double c)
	{
		if (a <= 0.0 || b <= 0.0 || c <= 0.0)
			throw domain_error("Sides cannot be 0 or less.");
		if (a > b + c || b > a + c || c > a + b)
			throw domain_error("One side cannot be larger than the sum of the others.");
	}

	triangle_kind kind(double a, double b, double c)
	{
		validate(a, b, c);

		if (a == b)
		{
			if (b == c)
			{
				return triangle_kind::equilateral;
			}
			else
			{
				return triangle_kind::isosceles;
			}
		}
		else if (a == c || b == c)
		{
			return triangle_kind::isosceles;
		}
		else
		{
			return triangle_kind::scalene;
		}
	}
}